benchmarks/dependencies_polars.py

```diff
@@ -0,0 +1,1010 @@
+                str,  # checksum
+                float,  # duration
+                str,  # format
+                int,  # removed
+                float,  # sampling_rate
+                int,  # type
+                str,  # version
+            ]
+        ],
+    ):
+        r"""Add media files.
+
+        Args:
+            values: list of tuples,
+                where each tuple holds the values of a new media entry
+
+        """
+        # pandas
+        # df = pd.DataFrame.from_records(
+        #     values,
+        #     columns=["file"] + list(define.DEPEND_FIELD_NAMES.values()),
+        # ).set_index("file")
+        # df.index = df.index.astype(define.DEPEND_INDEX_DTYPE)
+        # self._df = pd.concat([self._df, df])
+        self._df = pl.concat(
+            [self._df, pl.from_records(values, schema=self.schema_including_file)],
+            how="vertical_relaxed",
+        )
+
+    def _add_meta(
+        self,
+        file: str,
+        version: str,
+        archive: str,
+        checksum: str,
+    ):
+        r"""Add or update table file.
+
+        Args:
+            file: relative file path
+            archive: archive name without extension
+            checksum: checksum of file
+            version: version string
+
+        NOTE: casting to pd.DataFrame and back
+        """
+        format = audeer.file_extension(file).lower()
+
+        df = self.pl2pd(self._df)
+        # df = self._df.to_pandas().set_index(self.index_col)
+        df.loc[file] = [
+            archive,  # archive
+            0,  # bit_depth
+            0,  # channels
+            checksum,  # checksum
+            0.0,  # duration
+            format,  # format
+            0,  # removed
+            0,  # sampling_rate
+            define.DependType.META,  # type
+            version,  # version
+        ]
+        # self._df = pl.from_pandas(df, include_index=True)
+        self._df = self.pd2pl(df)
+
+    def _column_loc(
+        self,
+        column: str,
+        files: typing.Union[str, typing.Sequence[str]],
+        dtype: typing.Callable = None,
+    ) -> typing.Union[typing.Any, typing.List[typing.Any]]:
+        r"""Column content for selected files and column.
+
+        This implementation was slightly faster than this one
+        value = (
+            self._df.filter(self._df[self.index_col].is_in(files))[column].item()
+            if len(files) == 1
+            else self._df.filter(self._df[self.index_col].is_in(files))[
+                column
+            ].to_list()
+        )
+
+        """
+        value = (
+            self._df.filter(pl.col(self.index_col).is_in(files))
+            .select(pl.col(column))
+            .to_series()
+            .to_list()
+        )
+
+        if len(value) == 1:
+            return value
+
+        isscalar = not isinstance(value, Iterable)
+        if isscalar and dtype is not None:
+            value = dtype(value)
+
+        return value
```

Comment:
This implementation is very slow at the moment when requesting a single file. Is there maybe something similar to `df.at` with `polars` to speed this up?

Reply:
I have created an new paragraph with details on the new implementation [below](https://github.com/audeering/audb/pull/424#:~:text=1%20minute%20ago-,Treatment%20Index%20variable,-the%20previous%20previous).